staker/staker.gno

```diff
@@ -744,3 +750,78 @@ func getIncentives() map[string]ExternalIncentive {
 func getPoolIncentives() map[string][]string {
 	return poolIncentives
 }
+
+
+// TODO: separate into different file
+
+type StakerPoolInfo struct {
+	//totalStakedLiquidity *u256.Uint
+	totalStakedLiquidityInRange *u256.Uint
+}
+
+type StakerTickInfo struct {
+	liquidityInRangeDelta *u256.Uint
+}
+
+var (
+	stakerPoolInfo = make(map[string]StakerPoolInfo)
+	stakerTickInfo = make(map[int32]StakerTickInfo)
+)
+
+func modifyStakerPoolInfo(poolPath string, tokenId uint64, liquidity *u256.Uint) {
+	// update staker side pool info
+	pool, ok := stakerPoolInfo[poolPath]
+	if !ok {
+		pool = StakerPoolInfo{
+			totalStakedLiquidity: u256.Zero(),
+			totalStakedLiquidityInRange: u256.Zero(),
+		}
+	}
+	//pool.totalStakedLiquidity = liquidityMathAddDelta(pool.totalStakedLiquidity, liquidity)
+	if pn.PositionIsInRange(tokenId) {
+		pool.totalStakedLiquidityInRange = liquidityMathAddDelta(pool.totalStakedLiquidityInRange, liquidity)
```

Comment:
In order to calculate the reward correctly, we need to distinguish when tokenId is inrange and when it is not.
Currently, the history of tokenIds records all tokenIds in `deposits`. I would recommend that we consider an additional ledger, depositTokenId[poolPath][tokenId], that manages all tokenId leases deposited in poolPath, and an additional ledger, inRangeTokenIdList[poolPath][tokenId], that manages inRagneTokenIds. 
In some cases, it may not be an inRange, so you will need a global ledger and an inRange ledger.